### PR TITLE
Fix failures on Windows in hostname resource

### PR DESCRIPTION
**Pull request by johnmccrae**
Signed-off-by: John McCrae <jmccrae@chef.io>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).
Signed-off-by: John McCrae <jmccrae@chef.io>

### CODE REVIEW

File: lib/chef/resource/hostname.rb [hunk 1]
```diff
@@ -270,13 +270,24 @@ def updated_ec2_config_xml
 
           unless Socket.gethostbyname(Socket.gethostname).first == new_resource.hostname
             converge_by "set hostname to #{new_resource.hostname}" do
+              puts "What username is being used : #{new_resource.domain_user}"
```
Comment (mwrock) [inline lib/chef/resource/hostname.rb:273]: probably here by mistake?

File: lib/chef/resource/hostname.rb [hunk 2]
```diff
@@ -270,13 +270,24 @@ def updated_ec2_config_xml
...
-                  $user = #{new_resource.domain_user}
-                  $secure_password = #{new_resource.domain_password} | Convertto-SecureString -AsPlainText -Force
+                  $temp_user = #{new_resource.domain_user}
+                  $temp_password = #{new_resource.domain_password}
+                  if ([string]::IsNullOrEmpty($temp_user)){
+                    $temp_user = "Chef"
+                  }
+                  if ([string]::IsNullOrEmpty($temp_password)){
+                    $temp_user = "P@ssw0rd"
+                  }
```
Comment (mwrock) [inline lib/chef/resource/hostname.rb:288]: will this `chef` credential work if it does not exist? Seems like you at least need a comment here explaining.

File: tasks/rspec.rb [hunk 3]
```diff
@@ -43,29 +43,29 @@
 
   desc "Run all chef specs in spec directory"
   RSpec::Core::RakeTask.new(:spec) do |t|
-    t.verbose = false
+    t.verbose = true
```
Comment (tas50) [inline tasks/rspec.rb:46]: These need to get flipped back

**Comment by tas50** [comment]
Instead of setting fake credentials when they're necessary shouldn't we fail if the 2 properties aren't set and we need them?

**Review by mwrock — APPROVED** [review]

File: spec/functional/resource/windows_hostname_spec.rb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  describe "Changing machine names" do
+    context "The system can be renamed without a user or password when in a Workgroup" do
+      let(:hostname) { "Cucumber" }
+      it "does change" do
+        subject.run_action(:set)
+        expect(subject).to be_updated_by_last_action
+      end
+    end
+
+    context "the system gets renamed when in a domain" do
```
Comment (mwrock) [inline spec/functional/resource/windows_hostname_spec.rb:64]: Where are we domain joining?

Comment (mwrock) [reply spec/functional/resource/windows_hostname_spec.rb:64]: Oh the title of the context implies that the system is in a domain

**Comment by johnmccrae** [comment]
totally borked the tree and started a new PR to manage the update for this.

**Pull request closed without merging**